bazaarvoice/emodb
# Prune dependencies from emodb client

## Github Issue #

None

## What Are We Doing Here?

Our current client libraries are dependent on Jersey 1.x. This is has become an issue for many users as they cannot upgrade to Jersey 2. Additionally, if we upgrade our clients to Jersey 2, then users will be unable to use Jersey 1. This update attempts to remove the Jersey dependency altogether, and require the user to supply their own EmoClient.

This is still not complete for many reasons.
1. I have still yet to migrate our UAC client
2. I have not determined how to replace our client classes that were dependent ostrich, and therefore Jersey

## How to Test and Verify

No much to test yet, but you can see that I added a jax-rs EmoClient for Jersey 2.0 users to use. Additionally, users can use other http clients with it, such as Apache CXF or Restlet.

## Risk

### Level 

`High`

### Required Testing

`Smoke`, `Regression`, or `Manual`. (All changes except documentation need smoke
testing at a minimum).

### Risk Summary

The biggest risk of this change is that the new client does end up handling error responses as intended, and gives its user inaccurate responses.

## Code Review Checklist


build and run.

we inevitably come back to this code it will only take hours to figure out, not
days.

a victim of rampaging consistency, and should be using this course of action. 
We don't have coding standards out yet for this project, so please make sure to address any feedback regarding STYLE so the codebase remains consistent.
